test/libsolidity/semanticTests/experimental/stub.sol

```diff
@@ -1,5 +1,9 @@
 pragma experimental solidity;
 
+type word = __builtin("word");
+type bool = __builtin("bool");
```

Comment:
For booleans, it's funny that we could actually define them completely as user-defined type, i.e.
```
type bool = word;
```
Or later, more precisely, as an ADT
```
data bool = true | false;
```

The interesting part about booleans is rather that we have "truthiness" and "falsiness" on them, allowing them to work in conditionals.
Which is a bit of a chicken and egg problem: how to define truthiness without a boolean to indicate whether something is truthy :-)? So let's keep it builtin for now, but down the road it may be interesting to think about how to improve upon that.

Funnily, we wouldn't even actually need "truthiness"/"falsiness", if we defined ``if``s and loops in-language, which would actually work with higher-order functions and the ability to call Solidity functions from Yul if they take only types that involve "Yul-types" - but for that we'd among others need lambdas and what-not, so let's don't go there for now :-).

We might keep them as builtin permanently anyways, especially if we, in the future, introduce a boolean type in Yul (which would be one option for addressing stuff like https://github.com/ethereum/solidity/issues/13750, which I moved to the optimizer ideas hackmd, but would also complicate things quite a bit, so also nothing we'd target soon)